src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -578,42 +549,37 @@ public boolean isFeedbackSessionAttemptedByStudent(
     }
 
     /**
-     * Returns true if the feedback session is viewable by the given student.
+     * Returns true if the feedback session is viewable by the given user type.
      */
-    public boolean isFeedbackSessionViewableToStudents(
-            FeedbackSessionAttributes session) {
-        // Allow students to view the feedback session if there are questions for them
-        List<FeedbackQuestionAttributes> questionsToAnswer =
-                fqLogic.getFeedbackQuestionsForStudents(
-                        session.getFeedbackSessionName(), session.getCourseId());
-
-        if (session.isVisible() && !questionsToAnswer.isEmpty()) {
+    public boolean isFeedbackSessionViewableToUserType(FeedbackSessionAttributes session, boolean isInstructor) {
+        // Allow user to view the feedback session if there are questions for them
+        if (isFeedbackSessionForUserTypeToAnswer(session, isInstructor)) {
             return true;
         }
 
-        // Allow students to view the feedback session
-        // if there are any questions for instructors to answer
-        // where the responses of the questions are visible to the students
+        // Allow user to view the feedback session if there are any question whose responses are visible to the user
         List<FeedbackQuestionAttributes> questionsWithVisibleResponses = new ArrayList<>();
         List<FeedbackQuestionAttributes> questionsForInstructors =
```

Comment:
Should this variable be something along the lines of `questionsForUsers` or `feedbackQuestions` instead?